src/sys/ptrace.rs

```diff
@@ -182,14 +239,28 @@ pub fn syscall(pid: Pid) -> Result<()> {
 pub fn attach(pid: Pid) -> Result<()> {
     unsafe {
         ptrace_other(
-            ptrace::PTRACE_ATTACH,
+            Request::PTRACE_ATTACH,
             pid,
             ptr::null_mut(),
             ptr::null_mut(),
         ).map(|_| ()) // ignore the useless return value
     }
 }
 
+/// Detaches the current running process, as with `ptrace(PTRACE_DETACH, ...)`
+///
+/// Detaches from the process specified in pid allowing it to run freely
+pub fn detach(pid: Pid) -> Result<()> {
```

Comment:
Please move the addition of this function into a separate commit following the existing two and give it a separate entry under "Added" in the CHANGELOG.